### PR TITLE
Allow non-zero axis for .rolling (for application over columns)

### PR DESCRIPTION
**Pull request by mikegraham**

### CODE REVIEW

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -604,6 +606,7 @@ def rolling(self, window, min_periods=None, win_type=None):
 
         return Rolling(self, {'window': window,
                               'min_periods': min_periods,
+                              'axis': axis,
                               'win_type': win_type})
```
Comment (mrocklin) [inline dask/dataframe/core.py]: Thoughts on passing these in explicitly rather than as a dict?

File: dask/dataframe/rolling.py [hunk 2]
```diff
@@ -96,6 +96,9 @@ def __init__(self, obj, kwargs):
         self.obj = obj # dataframe or series
         self.rolling_kwargs = kwargs
```
Comment (mrocklin) [inline dask/dataframe/rolling.py]: ... and similarly assigning the fields directly to the object?

File: dask/dataframe/core.py (hunk 1 above)
Comment (mikegraham) [inline dask/dataframe/core.py]: I'm not entirely sure what it buys you other than having to type the same four or five fields several times and keep them synced up.

File: dask/dataframe/rolling.py (hunk 2 above)
Comment (mikegraham) [inline dask/dataframe/rolling.py]: This half I especially don't see the utility of. I must be missing something.

File: dask/dataframe/core.py (hunk 1 above)
Comment (mrocklin) [inline dask/dataframe/core.py]: OK, makes sense

**Comment by mrocklin** [comment]
Looks good to me.  Merging soon if no comments.

**Pull request merged**